tests/challenges/basic_abilities/test_write_file.py

```diff
@@ -1,41 +1,62 @@
-from typing import List
-
 import pytest
+from httpx import AsyncClient
 
-from autogpt.agent import Agent
-from autogpt.commands.file_operations import read_file
+from autogpt.workspace import Workspace
 from tests.challenges.challenge_decorator.challenge_decorator import challenge
-from tests.challenges.utils import get_workspace_path, run_interaction_loop
+from tests.challenges.utils import (
+    get_workspace_path,
+    setup_mock_input,
+    setup_mock_log_cycle_agent_name,
+)
 
 CYCLE_COUNT_PER_LEVEL = [1, 1]
 EXPECTED_OUTPUTS_PER_LEVEL = [
     {"hello_world.txt": ["Hello World"]},
     {"hello_world_1.txt": ["Hello World"], "hello_world_2.txt": ["Hello World"]},
 ]
+USER_REQUESTS = [
+    "Write 'Hello World' into a file named \"hello_world.txt\".",
+    'Write \'Hello World\' into 2 files named "hello_world_1.txt"and "hello_world_2.txt".',
+]
 
 
 @challenge()
-def test_write_file(
-    file_system_agents: List[Agent],
+async def test_write_file(
+    workspace: Workspace,
     patched_api_requestor: None,
     monkeypatch: pytest.MonkeyPatch,
     level_to_run: int,
     challenge_name: str,
+    client: AsyncClient,
 ) -> None:
-    file_system_agent = file_system_agents[level_to_run - 1]
-    run_interaction_loop(
-        monkeypatch,
-        file_system_agent,
-        CYCLE_COUNT_PER_LEVEL[level_to_run - 1],
-        challenge_name,
-        level_to_run,
+    user_request = USER_REQUESTS[level_to_run - 1]
+    setup_mock_input(monkeypatch, CYCLE_COUNT_PER_LEVEL[level_to_run - 1])
+    setup_mock_log_cycle_agent_name(monkeypatch, challenge_name, level_to_run)
+    response = await client.post(
+        "api/v1/agents",
+        json={"workspace": {"configuration": {"root": str(workspace.root)}}},
+    )
```

Comment:
we send the workspace when creating the agent, but we don't use that workspace in the create_agents controller.
This is because I am stuck in a loop: my only database is the workspace, so that's where I should put the agent_uuid, but at the same time this workspace location changes depending on the test !

I need a tiny sqlitedb for all the dumb crud endpoints. But I won't add it in this PR.